From 97114d86b828873568d9fc63a4ba92ff2721b2fd Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 11 Apr 2017 09:24:45 +0800
Subject: [PATCH 1660/5242] MLK-14655-3 video: mxsfb: copy timings info of
 primary fb to overlay fb

commit  cd1611d51f91ea3086cd0374853be816551ff09b from
https://source.codeaurora.org/external/imx/linux-imx.git

During overlay fb initialization, copy the timings information
of primary fb to init the timings of overlay fb.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit e47da3a65d32faec2fac7926b50980f0aa822343)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index eae5264..546a489 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1998,6 +1998,15 @@ static void init_mxsfb_overlay(struct mxsfb_info *fbi,
 	ofb->ol_fb->var.vmode		= FB_VMODE_NONINTERLACED;
 	ofb->ol_fb->var.nonstd		= 0;
 
+	/* Copy timings of primary fb */
+	ofb->ol_fb->var.pixclock	= fbi->fb_info->var.pixclock;
+	ofb->ol_fb->var.left_margin	= fbi->fb_info->var.left_margin;
+	ofb->ol_fb->var.right_margin	= fbi->fb_info->var.right_margin;
+	ofb->ol_fb->var.upper_margin	= fbi->fb_info->var.upper_margin;
+	ofb->ol_fb->var.lower_margin	= fbi->fb_info->var.lower_margin;
+	ofb->ol_fb->var.hsync_len	= fbi->fb_info->var.hsync_len;
+	ofb->ol_fb->var.vsync_len	= fbi->fb_info->var.vsync_len;
+
 	ofb->ol_fb->fbops = &overlay_fb_ops;
 	ofb->ol_fb->node  = -1;
 	ofb->ol_fb->par	  = ofb;
-- 
1.7.9.5

